nodestream/pipeline/transformers/transformer.py

```diff
@@ -99,6 +102,7 @@ def drain_completed_tasks():
                         submitted = True
                     for result in drain_completed_tasks():
```

Comment:
This code seems to repeat 3 times maybe something like this would work:
```
async def drain_completed_tasks_async(pending_tasks):
    for result in drain_completed_tasks():
        yield result
    await asyncio.sleep(0)
```

Then inside this function try to use await like in the comment above?
```
for result in await drain_completed_tasks_async(pending_tasks):
```